playlist.c

```diff
@@ -2769,6 +2780,55 @@ void playlist_get_db_name(playlist_t *playlist, size_t idx,
    }
 }
 
+core_info_t *playlist_entry_get_core(const struct playlist_entry* entry)
+{
+   core_info_ctx_find_t core_info;
+
+   core_info.inf = NULL;
+   core_info.path = entry->core_path;
+
+   if (!core_info_find(&core_info))
+      return NULL;
+
+   if (strcmp(core_info.inf->path, entry->core_path))
+   {
+      RARCH_LOG("[CORE] Found core '%s' path '%s' does not match requested path '%s'", core_info.inf->display_name, core_info.inf->path, entry->core_path);
+   }
+
+   return core_info.inf;
+}
+
+core_info_t *playlist_get_default_core(playlist_t* playlist)
+{
+   core_info_ctx_find_t core_info;
+   const char* default_core_path =
+      playlist_get_default_core_path(playlist);
+   const char* default_core_name =
+      playlist_get_default_core_path(playlist);
+
+   if (!default_core_path && !default_core_name)
+      return NULL;
+
+   char new_core_path[PATH_MAX_LENGTH];
+   new_core_path[0] = '\0';
+
+   if (!string_is_empty(default_core_path))
+   {
+      strlcpy(new_core_path, default_core_path, sizeof(new_core_path));
+      playlist_resolve_path(PLAYLIST_LOAD, new_core_path, sizeof(new_core_path));
+   }
+
+   core_info.inf = NULL;
```

Comment:
Can assign `NULL` when initialising the variable - it's a bit ugly to do it here

Reply:
Sure, but I found SO many of these around the code ;)